File: client/src/types/registration.ts
export const CHESS_LEVELS = [
  'Fillestar',
  'Elementar',
  'Mesatar',
  'Avancuar',
  'Ekspert',
] as const;

export interface Registration {
  id: number;
  kidFullName: string;
  kidAge: number;
  kidSchool: string;
  kidChessLevel: string;
  parentName: string;
  parentPhone: string;
  parentEmail: string;
  createdAt: string;
  updatedAt: string | null;
}

export interface RegistrationFormData {
  kidFullName: string;
  kidAge: number | '';
  kidSchool: string;
  kidChessLevel: string;
  parentName: string;
  parentPhone: string;
  parentEmail: string;
}

export const emptyForm: RegistrationFormData = {
  kidFullName: '',
  kidAge: '',
  kidSchool: '',
  kidChessLevel: 'Fillestar',
  parentName: '',
  parentPhone: '',
  parentEmail: '',
};

File: ChessCampRegistration.Api/Models/Registration.cs
namespace ChessCampRegistration.Api.Models;

public class Registration
{
    public int Id { get; set; }
    public required string KidFullName { get; set; }
    public int KidAge { get; set; }
    public required string KidSchool { get; set; }
    public required string KidChessLevel { get; set; }
    public required string ParentName { get; set; }
    public required string ParentPhone { get; set; }
    public required string ParentEmail { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}
